### PR TITLE
- added possibility to use files from the file system

### PR DESCRIPTION
**Pull request by ruerob**
Hello @yang-f,

I needed to read SVG files from the devices file system. So I added a small bit of code to your work here. I hope it is okay and can be used in this way. I was unsure whether to use the ternary operator or an if branch.

Thank you for building this fine small piece of code and have a nice day
Robert

### CODE REVIEW

File: lib/flutter_svg_provider.dart [hunk 1]
```diff
@@ -19,8 +20,8 @@ import 'package:flutter_svg/flutter_svg.dart';
 /// )
 /// ```
 class Svg extends ImageProvider<SvgImageKey> {
-  /// Path to svg file asset
-  final String asset;
+  /// Path to svg file or asset
+  final String filePath;
```
Comment (yang-f) [inline lib/flutter_svg_provider.dart:24]: rename this variable to `path`?

Comment (ruerob) [reply lib/flutter_svg_provider.dart]: Yepp! Done! :+1:

File: lib/flutter_svg_provider.dart [hunk 2]
```diff
@@ -56,12 +60,15 @@ class Svg extends ImageProvider<SvgImageKey> {
...
     return OneFrameImageStreamCompleter(
-      _loadAsync(key),
+      _loadAsync(key, isAsset),
     );
   }
 
-  static Future<ImageInfo> _loadAsync(SvgImageKey key) async {
-    final String rawSvg = await rootBundle.loadString(key.assetName);
+  static Future<ImageInfo> _loadAsync(SvgImageKey key, bool isAsset) async {
+    String rawSvg = (isAsset)?
```
Comment (yang-f) [inline lib/flutter_svg_provider.dart:68]: Should use `dartfmt` to format code.

File: lib/flutter_svg_provider.dart [hunk 3]
```diff
@@ -70,7 +77,7 @@ class Svg extends ImageProvider<SvgImageKey> {
       ),
       clipToViewBox: false,
       colorFilter:
-          ColorFilter.mode(key.color ?? Colors.transparent, BlendMode.srcATop),
+      ColorFilter.mode(key.color ?? Colors.transparent, BlendMode.srcATop),
```
Comment (yang-f) [inline lib/flutter_svg_provider.dart:80]: formart

File: lib/flutter_svg_provider.dart (hunk 2 above)
Comment (yang-f) [reply lib/flutter_svg_provider.dart:68]: like this:
```dart
-    String rawSvg = (isAsset)?
-    await rootBundle.loadString(key.assetName):
-    await File(key.assetName).readAsString();
+    String rawSvg = (isAsset)
+        ? await rootBundle.loadString(key.assetName)
+        : await File(key.assetName).readAsString();
```

File: lib/flutter_svg_provider.dart (hunk 3 above)
Comment (yang-f) [reply lib/flutter_svg_provider.dart:80]:
```dart
-      ColorFilter.mode(key.color ?? Colors.transparent, BlendMode.srcATop),
+          ColorFilter.mode(key.color ?? Colors.transparent, BlendMode.srcATop),
```

File: lib/flutter_svg_provider.dart (hunk 2 above)
Comment (ruerob) [reply lib/flutter_svg_provider.dart:68]: Ah! Okay! Haha! Done in a minute.

File: lib/flutter_svg_provider.dart (hunk 3 above)
Comment (ruerob) [reply lib/flutter_svg_provider.dart:80]: Here I think the tabbed one was the correct one because it is the value of colorFilter, isn't it?

Comment (ruerob) [reply lib/flutter_svg_provider.dart]: Sorry! I've read it wrong. It's my first pull request where I've gotten annotations! Haha!

**Review by yang-f — APPROVED** [review]

**Pull request merged**